vm/parser.ts

```diff
@@ -711,6 +861,17 @@ module J2ME {
       }
     }
 
+    getClassNameSlow(): string {
+      return this._name || (this._name = ByteStream.readString(this.utf8Name));
+    }
```

Comment:
My goal is to convert all accesses to classInfo.className, methodInfo.name and methodInfo.signature to getXXX methods. To keep the size of the patch down, I used getters for name and signature so I wouldn't have to update all the call sites.

I tried a quick benchmark, and the getXXX methods are the fastest, probably because they get inlined and have monomorphic call sites:

Getter: 216.31
Memoizer: 3.94
Call: 0.92

Getters are obviously really bad.

```

function foo() {
	var s = 0;
	for (var i = 0; i < 1000; i++) {
		var o = {_x: 123};	
		Object.defineProperty(o, "x", {
			get: function () {
				return this._x;
			}
		})
		for (var j = 0; j < 1000; j++) {
			s += o.x;
		}
	}
}

function bar() {
	var s = 0;
	for (var i = 0; i < 1000; i++) {
		var o = {_x: 123};	
		Object.defineProperty(o, "x", {
			get: function () {
				Object.defineProperty(o, "x", {value: this._x});
				return this._x;
			}, configurable: true
		})
		for (var j = 0; j < 1000; j++) {
			s += o.x;
		}
	}
}

function car() {
	var s = 0;
	for (var i = 0; i < 1000; i++) {
		var o = {_x: 123, getX: function () {
			return this._x;
		}};	
		for (var j = 0; j < 1000; j++) {
			s += o.getX();
		}
	}
}

var s = dateNow();
foo();
print("Getter: " + (dateNow() - s).toFixed(2));

var s = dateNow();
bar();
print("Memoizer: " + (dateNow() - s).toFixed(2));

var s = dateNow();
car();
print("Call: " + (dateNow() - s).toFixed(2));
```